src/komponenter/kalkulator/Fraværsperioder/Fraværsperiode.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { FunctionComponent, useState } from 'react';
+import { AllePermitteringerOgFraværesPerioder, DatoIntervall } from '../typer';
+import DatoIntervallInput from '../DatointervallInput/DatointervallInput';
+import { datoIntervallErGyldig, fraværInngårIPermitteringsperioder } from '../utils/dato-utils';
+
+interface Props {
+    indeks: number;
+    allePermitteringerOgFraværesPerioder: AllePermitteringerOgFraværesPerioder;
+    setAllePermitteringerOgFraværesPerioder: (
+        allePermitteringerOgFraværesPerioder: AllePermitteringerOgFraværesPerioder
+    ) => void;
+}
+
+const Fraværsperiode: FunctionComponent<Props> = ({ indeks, allePermitteringerOgFraværesPerioder, setAllePermitteringerOgFraværesPerioder
+}) => {
+    const [uvalidertDatoIntervall, setUvalidertDatoIntervall] = useState<
+        Partial<DatoIntervall>
+        >({ datoFra: undefined, datoTil: undefined });
+
+    const [
+        advarselFraværsperiodeUtenforPermittering,
+        setAdvarselFraværsperiodeUtenforPermittering,
+    ] = useState<string>('');
+
+    const oppdaterDatoIntervall = (
+        datoIntervall: Partial<DatoIntervall>
+    ) => {
+        setUvalidertDatoIntervall(datoIntervall)
+        if (datoIntervallErGyldig(datoIntervall)) {
+            if (
+                datoIntervall.datoFra &&
+                !fraværInngårIPermitteringsperioder(
+                    allePermitteringerOgFraværesPerioder.permitteringer,
+                    datoIntervall
+                )
+            ) {
+                setAdvarselFraværsperiodeUtenforPermittering(
+                    'Fravær som ikke inngår i permitteringsperioder telles ikke med i beregningen'
+                );
+            } else {
+                setAdvarselFraværsperiodeUtenforPermittering('');
+            }
+            const kopiAvFraværsperioder = [
+                ...allePermitteringerOgFraværesPerioder.andreFraværsperioder,
+            ];
+            kopiAvFraværsperioder[indeks] = datoIntervall;
+            setAllePermitteringerOgFraværesPerioder({
+                ...allePermitteringerOgFraværesPerioder,
+                andreFraværsperioder: kopiAvFraværsperioder,
+            });
+        }
+    };
+
+    const slettFraværsperiode = () => {
+        const kopiAvAllPermitteringsInfo = {
+            ...allePermitteringerOgFraværesPerioder,
+        };
+        if (kopiAvAllPermitteringsInfo.andreFraværsperioder.length > 0) {
+            kopiAvAllPermitteringsInfo.andreFraværsperioder.splice(indeks, 1);
+        }
+        setAllePermitteringerOgFraværesPerioder(
+            kopiAvAllPermitteringsInfo
+        );
+    };
+
+    return (
+        <DatoIntervallInput
+            advarsel={advarselFraværsperiodeUtenforPermittering}
+            key={indeks}
+            datoIntervall={
+                uvalidertDatoIntervall
```

Comment:
Her kopieres den globale staten til en lokal state, og det er den lokale staten som vises. Hvis disse to statene kommer i usync på en eller annen måte, så kan det gi bugs som er skikkelig hodepine å oppdage. Anbefaler å bruke den globale staten i stedet, selv om det kommer et ugyldig datointervall. Vi kan heller fange det opp på andre måter, som å returnere `undefined` i funksjonen som konverterer Partial-intervaller til definerte intervaller. Evt. gjøre en sjekk på at datointervallet er gyldig der det trengs.